src/main/java/com/todo/backend/Controller.java

```diff
@@ -1,54 +1,302 @@
+                if (due_date != null) {
+                    if (due_date.equals("none"))
+                        todo.setDue_date(null);
+                    else
+                        todo.setDue_date(Instant.parse(due_date));
+                }
+
+                this.metrics.needsRecalculate();
+                this.lastResponse = null; // FORCE RECALCULATE
+                return new ResponseEntity<>("Item modified", HttpStatus.OK);
+            }
+        }
+
+        return new ResponseEntity<>("Not Found", HttpStatus.NOT_FOUND);
+    }
+    // #endregion
+
+    // #region ################################ DONE/UNDONE
+    @RequestMapping(method = { RequestMethod.POST }, value = { "/todos/{id}/done" })
+    public ResponseEntity<String> markDone(@PathVariable int id, @RequestParam(required = false) Integer inPage) {
+        if (id < 0)
+            return new ResponseEntity<>("Invalid ID", HttpStatus.BAD_REQUEST);
+
+        return this.toggleDoneUndone(id, inPage, true);
+    }
+
+    @RequestMapping(method = { RequestMethod.PUT }, value = { "/todos/{id}/undone" })
+    public ResponseEntity<String> markUndone(@PathVariable int id, @RequestParam(required = false) Integer inPage) {
+        if (id < 0)
+            return new ResponseEntity<>("Invalid ID", HttpStatus.BAD_REQUEST);
+
+        return this.toggleDoneUndone(id, inPage, false);
+    }
+    // #endregion
+
+    // #region ################################ DELETE
+    @RequestMapping(method = { RequestMethod.DELETE }, value = { "/todos/{id}/delete" })
+    public ResponseEntity<String> deleteToDo(@PathVariable int id) {
+        if (id < 0)
+            return new ResponseEntity<>("Invalid ID", HttpStatus.BAD_REQUEST);
+
+        boolean found = DB.removeIf(todo -> todo.getId() == id);
+
+        this.metrics.needsRecalculate();
+        this.lastResponse = null; // FORCE RECALCULATE
 
-    public Controller() {
-        items.add("initial value");
+        if (found)
+            return new ResponseEntity<>("Deleted", HttpStatus.OK);
+        return new ResponseEntity<>("Not Found", HttpStatus.NOT_FOUND);
     }
+    // #endregion
+
+    /**
+     * Shortcut to save previous Search Params and see if its the same or not
+     */
+    public class SearchParams {
+        private Boolean done = null;
+        private String text = null;
+        private Priority priority = null;
+        private Boolean sortPriority = null;
+        private Boolean sortDueDate = null;
+
+        public Boolean isSameAsPrevious(
+                Boolean done,
+                String text,
+                Priority priority,
+                Boolean sortPriority,
+                Boolean sortDueDate) {
+            Boolean check = true;
+
+            if (done != this.done || text != this.text || priority != this.priority
+                    || sortPriority != this.sortPriority || sortDueDate != this.sortDueDate) {
+                check = false;
+            }
+
+            this.done = done;
```

Comment:
the name of the method reflects a validation but not a change in the state, you could do keep name `isSameAsPrevious` and use another method call to set state values separately

Reply:
what about `updateIfDiffers`